### PR TITLE
Feature/7426 aca infra

### PR DESCRIPTION
**Pull request by satenderrathee**
📲 What
Created new project for Azure Container Apps.
Created Azure Container Apps using module.
Added a sample application and enabled its deployment.
The deployment of ACA using Terraform with Ensono Independent Runner (EIR).

🤔 Why
Stack cycle 14
7294- Create Azure Container Apps Terraform module

🛠 How
Added new project for Azure Container Apps, used terraform module to provision ACA and enable its deployment using Independent Runner.

👀 Evidence
Pipeline:
image

image

image

### CODE REVIEW

File: src/terraform/data.tf
Comment (danEnsono) [inline src/terraform/data.tf:1]: I think all of this terraform should still be in the `deploy\terraform` directory, perhaps under a different folder such as `additional-infra` with your primary terraform referenced as `base-infra`. That way we aren't spreading the deployment code out and it remains in the places pre-defined for deployment type code.

File: src/terraform/data.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+data "azurerm_client_config" "current" {}
+
+data "azurerm_container_registry" "acr" {
+  name                = var.acr_name
+  resource_group_name = var.acr_rg #"aca-test-rg"  module.default_label.id
```
Comment (danEnsono) [inline src/terraform/data.tf:5]: Remove comments

File: src/terraform/locals.tf
Comment (danEnsono) [inline src/terraform/locals.tf:1]: I think all of this terraform should still be in the `deploy\terraform` directory, perhaps under a different folder such as `additional-infra` with your primary terraform referenced as `base-infra`. That way we aren't spreading the deployment code out and it remains in the places pre-defined for deployment type code.

File: src/terraform/main.tf
Comment (danEnsono) [inline src/terraform/main.tf:1]: I think all of this terraform should still be in the `deploy\terraform` directory, perhaps under a different folder such as `additional-infra` with your primary terraform referenced as `base-infra`. That way we aren't spreading the deployment code out and it remains in the places pre-defined for deployment type code.

File: src/terraform/outputs.tf
Comment (danEnsono) [inline src/terraform/outputs.tf:1]: I think all of this terraform should still be in the `deploy\terraform` directory, perhaps under a different folder such as `additional-infra` with your primary terraform referenced as `base-infra`. That way we aren't spreading the deployment code out and it remains in the places pre-defined for deployment type code.

File: src/terraform/provider.tf
Comment (danEnsono) [inline src/terraform/provider.tf:1]: I think all of this terraform should still be in the `deploy\terraform` directory, perhaps under a different folder such as `additional-infra` with your primary terraform referenced as `base-infra`. That way we aren't spreading the deployment code out and it remains in the places pre-defined for deployment type code.

File: src/terraform/variables.tf
Comment (danEnsono) [inline src/terraform/variables.tf:1]: I think all of this terraform should still be in the `deploy\terraform` directory, perhaps under a different folder such as `additional-infra` with your primary terraform referenced as `base-infra`. That way we aren't spreading the deployment code out and it remains in the places pre-defined for deployment type code.

File: taskctl.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  #     depends_on: infra:plan
+
+  # infrastructure_destroy:
+  #   - task: infra:init
+  #   - task: infra:destroy:plan
+  #     depends_on:
+  #       - infra:init
+  #   - task: infra:destroy:apply
+  #     depends_on:
+  #       - infra:destroy:plan
```
Comment (danEnsono) [inline taskctl.yaml:31]: Remove this

File: taskctl.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    - task: build:container
+
+  setup:
+    - task: build:number
+
+  # setup:
+  #   - task: debug:env
+  #   - task: buildnumber
+  #     depends_on:
+  #       - debug:env
```
Comment (danEnsono) [inline taskctl.yaml:77]: Remove this

File: taskctl.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  docs:
+    - task: build:number
+    - task: _docs
+      depends_on: build:number
+
+  release:
+    - task: _release
+
+  build:container:
+    - task: debug
```
Comment (danEnsono) [inline taskctl.yaml:67]: Do you always want to run debug for build:container task?

**Review by danEnsono — CHANGES_REQUESTED** [review]
Good overall work, some rethinking required around directory structure and redundant variable files. Lots of left over comments to remove and a few lock files.

Needs some decent documentation within the repo to support the feature outlining:
Why are we doing this feature?
How do you use this feature?
What is the pipeline doing?
What are the tests doing?

File: src/terraform/variables.tf
Comment (satenderrathee) [reply src/terraform/variables.tf:1]: These are App specific Infra , so kept under app, we have used the same approach in stack data project as well. I have mentioned that in agreed ADR as well https://amidodevelopment.atlassian.net/wiki/spaces/TEC/pages/edit-v2/4508418074?draftShareId=74f0c4c4-0ebf-4e5b-94bf-e99d8d8403dd.

File: src/terraform/provider.tf
Comment (satenderrathee) [reply src/terraform/provider.tf:1]: These are App specific Infra , so kept under app, we have used the same approach in stack data project as well. I have mentioned that in agreed ADR as well https://amidodevelopment.atlassian.net/wiki/spaces/TEC/pages/edit-v2/4508418074?draftShareId=74f0c4c4-0ebf-4e5b-94bf-e99d8d8403dd.

File: taskctl.yaml (hunk 4 above)
Comment (satenderrathee) [reply taskctl.yaml]: updated.

File: taskctl.yaml (hunk 3 above)
Comment (satenderrathee) [reply taskctl.yaml]: removed

File: taskctl.yaml (hunk 2 above)
Comment (satenderrathee) [reply taskctl.yaml]: removed

File: cicd/build/azDevOps/azure/deploy-infrastructure.yml [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+                            
+                    #         TF_FILE_LOCATION: /app/deploy/terraform/azure
+                    #         TF_VAR_name_environment: ${{ stage.environment_shortname }}
+                    #         TF_VAR_location: $(region)
+                    #         TF_VAR_resource_group_name: $(resource_group_name)
+
+                    #         AZURE_CLIENT_ID: $(azure-client-id)
+                    #         AZURE_CLIENT_SECRET: $(azure-client-secret)
+                    #         AZURE_SUBSCRIPTION_ID: $(azure-subscription-id)
+                    #         AZURE_TENANT_ID: $(azure-tenant-id)
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/deploy-infrastructure.yml]: removed

File: cicd/build/azDevOps/azure/infrastructure-aca-vars.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  - name: team_owner
+    value: plateng
+
+  # SelfConfig
+  # If you haven't specified source_repo at cli runtime please ensure you replace it here
+  # It is case sensitive for TFS based repos
+  # - name: self_repo_src
+  #   value: src/api
+  - name: self_repo_dir
+    value: "$(Agent.BuildDirectory)/s/$(self_repo)"
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/infrastructure-aca-vars.yml]: added required variables in  pipeline-vars.yml

File: cicd/build/azDevOps/azure/pipeline-vars.template.yml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  # there are some best practices around this if you are going for feature based environments
+  # avoid running anything past dev that is not on master
+  - name: tf_state_rg
+    value: {{ .Input.Terraform.Backend.Group }}
+  - name: tf_state_storage
+    value: {{ .Input.Terraform.Backend.Storage }}
+  - name: tf_state_container
+    value: {{ .Input.Terraform.Backend.Container }}
+  - name: tf_state_key
+    value: core
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.template.yml:33]: removed

File: cicd/build/azDevOps/azure/pipeline-vars.template.yml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  - name: create_dns_zone
+    value: false
+
+  # ------- GitHub
+  # - name: create_release
+  #   value: false
+  # - name: github_release_service_connection
+  #   value:
+  # - name: github_org
+  #   value: $(company)
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.template.yml]: uncommented

File: cicd/build/azDevOps/azure/pipeline-vars.template.yml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+  - name: aks_cluster_version
+    value: 1.24.9
+
+  # Should a valid certificate be obtained, or use a self signed one
+  - name: create_valid_cert
+    value: false
+
+  # Additional node pools for kubernetes
+  - name: aks_node_pools
+    value: {}
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.template.yml]: removed

File: cicd/build/azDevOps/azure/pipeline-vars.template.yml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    value: {{ .Input.Business.Domain }}
+
+  # the email address to use when creating the SSL certificate
+  - name: acme_email
+    value:
+
+  # -------- Cluster information
+  # version of AKS to deploy
+  - name: aks_cluster_version
+    value: 1.24.9
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.template.yml]: removed

File: cicd/build/azDevOps/azure/pipeline-vars.yml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  # there are some best practices around this if you are going for feature based environments
+  # avoid running anything past dev that is not on master
+  - name: tf_state_rg
+    value: Stacks-Ancillary-Resources #stacks-terraform-state
+  - name: tf_state_storage
+    value: amidostackstfstate #stacksstatehjfis
+  - name: tf_state_container
+    value: tfstate
+  # - name: tf_state_key
+  #   value: acacore
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.yml]: removed

File: cicd/build/azDevOps/azure/pipeline-vars.yml [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+  # ------- GitHub
+  # - name: create_release
+  #   value: false
+  # - name: github_release_service_connection
+  #   value: GitHubReleases
+  # - name: github_org
+  #   value: $(company)
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.yml]: done

File: cicd/build/azDevOps/azure/pipeline-vars.yml [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+  - name: TaskctlVersion
+    value: 1.5.2
+
+  # Set the name of the cloud platform
+  - name: CLOUD_PLATFORM
+    value: azure
+
+
+  # - name: resource_group_name
+  #   value: ed-stacks-nonprod-uks-aca
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.yml]: done

File: cicd/build/azDevOps/templates/infra-tests.yml [hunk 14]
```diff
@@ -0,0 +1,35 @@
+parameters:
+  INSPEC_FILES:
+  INSPEC_ARGS:
+  INSPEC_OUTPUT_PATH: /app/outputs/tests
+  CHEF_LICENSE:
+  REPORT_FILENAME:
+  TF_FILE_LOCATION:
+  TF_VAR_name_environment:
+  TF_VAR_location:
+  # TF_VAR_resource_group_name:
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/templates/infra-tests.yml:10]: done

File: cicd/build/azDevOps/templates/infra-tests.yml [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+      INSPEC_FILES: ${{ parameters.inspec_files }}
+      CHEF_LICENSE: ${{ parameters.chef_license }}
+      INSPEC_ARGS: ${{ parameters.inspec_args }}
+      INSPEC_OUTPUT_PATH: ${{ parameters.inspec_output_path }}
+      REPORT_FILENAME: ${{ parameters.report_filename }}
+
+      TF_FILE_LOCATION: ${{ parameters.tf_file_location }}
+      TF_VAR_name_environment: ${{ parameters.tf_var_name_environment }}
+      TF_VAR_location: ${{ parameters.TF_VAR_location }}
+      # TF_VAR_resource_group_name: ${{ parameters.TF_VAR_resource_group_name }}
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/templates/infra-tests.yml]: done

File: cicd/build/config/stage_envvars.yml [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+      # - name: TF_VAR_location
+      #   description: The location of the resource group
+      # - name: TF_VAR_resource_group_name
+      #   description: The name of the resource group where the Azure Container Group and other new components (ACR, VNET) should be created or where the existing ACR is located.
+      # - name: TF_VAR_create_acr
+      #   description: A boolean flag indicating whether to create a new Azure Container Registry (ACR). If set to true, a new ACR will be created; if set to false, an existing ACR will be used.
+      # - name: TF_VAR_acr_name
+      #   description: The name of the new or existing Azure Container Registry (ACR).
+      # - name: TF_VAR_acr_location
+      #   description: The location/region where the Azure Container Registry (ACR) should be created or where the existing ACR is located.
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/config/stage_envvars.yml]: done

File: cicd/build/scripts/Deploy-Charts.ps1
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/scripts/Deploy-Charts.ps1:1]: done

File: cicd/build/taskctl/tasks.yaml [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+  #   command:
+  #     - |
+  #       Import-Module Az.ContainerInstance
+  #       # TODO: Remove below for now until correct outputs from the TF Module itself
+  #       # Invoke-Terraform -Workspace -Arguments $env:TF_VAR_name_environment -Path $env:TF_FILE_LOCATION
+  #       # Invoke-Terraform -Output -Path $env:TF_FILE_LOCATION | /app/cicd/build/scripts/Set-EnvironmentVars.ps1 -prefix "TFOUT" -key "value" -passthru | ConvertTo-Yaml | Out-File -Path ${PWD}/inspec_inputs.yml
+  #       Set-Content -Path "${PWD}/inspec_inputs.yml" -Value ""
+  #       Add-Content -Path ${PWD}/inspec_inputs.yml -Value "resource_group_name: ${env:TF_VAR_resource_group_name}"
+  #       Add-Content -Path ${PWD}/inspec_inputs.yml -Value "location: ${env:TF_VAR_location}"
+  #       Invoke-Inspec -execute -arguments "--input-file ${PWD}/inspec_inputs.yml" -path $env:INSPEC_FILES
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/taskctl/tasks.yaml]: done

File: deploy/terraform/azure/outputs.tf [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+  value = module.acae.container_app_environment_id
+}
+
+# output "acr_name" {
+#   value = azurerm_container_registry.acr.name
+# }
+
+# output "acr_id" {
+#   value = azurerm_container_registry.acr.id
+# }
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply deploy/terraform/azure/outputs.tf]: done

File: cicd/build/azDevOps/azure/deploy-infrastructure.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/deploy-infrastructure.yml]: removed

File: cicd/build/taskctl/tasks copy.yaml
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/taskctl/tasks copy.yaml:1]: done

File: src/deploy-aca.yml [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+
+          - task: Bash@3
+            displayName: "TaskCTL: Build Image"
+            inputs:
+              targetType: inline
+              script: |
+                taskctl build:container
+            env:
+
+              #SELF_REPO_SRC: src/app
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: updated

File: cicd/build/azDevOps/azure/deploy-infrastructure.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/deploy-infrastructure.yml]: done

File: deploy/terraform/azure/provider.tf [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.108.0"
+    }
+  }
+  backend "azurerm" {
+    # resource_group_name  = "ResourceGroupName"  # Name of the resource group that your storage account resides in.
+    # storage_account_name = "StorageAccountName" # Name of the storage account for your terraform state file.
+    # container_name       = "tfstate"            # What your container name within the storage account is called.
+    # key                  = "terraform.tfstate"  # What your state output will be named.
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply deploy/terraform/azure/provider.tf:13]: done

File: src/terraform/outputs.tf
Comment (satenderrathee) [reply src/terraform/outputs.tf:1]: These are App specific Infra , so kept under app, we have used the same approach in stack data project as well. I have mentioned that in agreed ADR as well https://amidodevelopment.atlassian.net/wiki/spaces/TEC/pages/edit-v2/4508418074?draftShareId=74f0c4c4-0ebf-4e5b-94bf-e99d8d8403dd.

File: src/terraform/main.tf
Comment (satenderrathee) [reply src/terraform/main.tf:1]: These are App specific Infra , so kept under app, we have used the same approach in stack data project as well. I have mentioned that in agreed ADR as well https://amidodevelopment.atlassian.net/wiki/spaces/TEC/pages/edit-v2/4508418074?draftShareId=74f0c4c4-0ebf-4e5b-94bf-e99d8d8403dd.

File: src/terraform/locals.tf
Comment (satenderrathee) [reply src/terraform/locals.tf:1]: These are App specific Infra , so kept under app, we have used the same approach in stack data project as well. I have mentioned that in agreed ADR as well https://amidodevelopment.atlassian.net/wiki/spaces/TEC/pages/edit-v2/4508418074?draftShareId=74f0c4c4-0ebf-4e5b-94bf-e99d8d8403dd.

File: src/terraform/data.tf (hunk 1 above)
Comment (satenderrathee) [reply src/terraform/data.tf:5]: done

File: src/terraform/data.tf
Comment (satenderrathee) [reply src/terraform/data.tf:1]: These are App specific Infra , so kept under app, we have used the same approach in stack data project as well. I have mentioned that in agreed ADR as well https://amidodevelopment.atlassian.net/wiki/spaces/TEC/pages/edit-v2/4508418074?draftShareId=74f0c4c4-0ebf-4e5b-94bf-e99d8d8403dd.

File: deploy/terraform/azure/variables.tf [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+variable "name_component" {
+  description = "Component Name - should/will be used in conventional resource naming. Typically this will be a logical name for this part of the system i.e. `API` || `middleware` or more generic like `Billing`"
+  type        = string
+  #default     = "shared"
+}
+
+variable "stage" {
+  type = string
+  #default = "dev"
+}
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply deploy/terraform/azure/variables.tf]: done

File: deploy/tests/inspec.lock
[the remark this answers is not in the document]
Comment (satenderrathee) [reply deploy/tests/inspec.lock:1]: added

File: src/deploy-aca.yml [hunk 22]
```diff
@@ -0,0 +1,314 @@
...
+          #     targetType: inline
+          #     script: |
+          #       taskctl docs
+
+          # # Upload the documentation as a build artifact
+          # - task: PublishBuildArtifacts@1
+          #   displayName: Publish Documentation
+          #   inputs:
+          #     pathToPublish: $(Build.SourcesDirectory)/outputs/docs
+          #     artifactName: docs
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: removed

File: src/terraform/.terraform.lock.hcl
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/terraform/.terraform.lock.hcl:1]: added

File: cicd/build/azDevOps/azure/pipeline-vars.template.yml
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/pipeline-vars.template.yml:1]: done

File: src/aca-vars.yml
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/aca-vars.yml:1]: This is app specific variable , we are not targeting Stacks-CLI to output app specific variables . So kept them separate.

File: cicd/build/azDevOps/azure/deploy-infrastructure.yml [hunk 23]
```diff
@@ -0,0 +1,304 @@
...
+
+
+  # Called Non-Prod instead of Dev as all Non-Production stages will deploy to here
+  - ${{ each stage in parameters.stages }}:
+      - stage: ${{ stage.stage }}
+        dependsOn: ${{ stage.dependsOn }}
+        condition: and(succeeded(), or(${{ stage.condition }}, ${{ parameters.force_deploy }}))
+        variables:
+          # Pull in the non-production credentials for the build to use
+          - group: stacks-credentials-nonprod-kv #ensono-sp-creds
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/deploy-infrastructure.yml]: done

File: cicd/build/azDevOps/azure/deploy-infrastructure.yml [hunk 24]
```diff
@@ -0,0 +1,304 @@
...
+                    - template: ../templates/setup.yml
+                      parameters:
+                        TaskctlVersion: ${{ variables.TaskctlVersion }}
+
+                    - bash: |
+                        echo "##vso[task.setvariable variable=ARM_TENANT_ID]$(azure-tenant-id)"
+                        echo "##vso[task.setvariable variable=ARM_SUBSCRIPTION_ID]$(azure-subscription-id)"
+                        echo "##vso[task.setvariable variable=ARM_CLIENT_ID]$(azure-client-id)"
+                        echo "##vso[task.setvariable variable=ARM_CLIENT_SECRET]$(azure-client-secret)"
+                      displayName: Set Evn Variables
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/deploy-infrastructure.yml]: removed

File: cicd/build/azDevOps/azure/deploy-infrastructure.yml [hunk 25]
```diff
@@ -0,0 +1,304 @@
...
+
+  - stage: Release
+    dependsOn:
+      - ${{ if ne(variables['Build.SourceBranch'], 'refs/heads/master') }}:
+          - NonProd
+      - ${{ else }}:
+          - Prod
+    condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(${{ parameters.pre_release }}, true)))
+    variables:
+      - group: amido-stacks-github-credentials   #amido-stacks-webapp
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/azDevOps/azure/deploy-infrastructure.yml]: done

File: src/deploy-aca.yml [hunk 26]
```diff
@@ -0,0 +1,314 @@
...
+    value: $[counter(join(variables['version_major'], join('-', variables['version_minor'])), 0)]
+  - name: version_number
+    value: "$(version_major).$(version_minor).$(version_revision)-${{ lower(variables['Build.SourceBranchName']) }}"
+  # - name: DOCKER_IMAGE_TAG
+  #   value: "$(version_number)"
+
+stages:
+  - stage: Build
+    variables:
+      - group: stacks-credentials-nonprod-kv
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: done

File: src/deploy-aca.yml [hunk 27]
```diff
@@ -0,0 +1,314 @@
...
+          #     artifactName: docs
+
+  # Called Non-Prod instead of Dev as all Non-Production stages will deploy to here
+  - ${{ each stage in parameters.stages }}:
+      - stage: ${{ stage.stage }}
+        dependsOn: ${{ stage.dependsOn }}
+        condition: and(succeeded(), or(${{ stage.condition }}, ${{ parameters.force_deploy }}))
+        variables:
+          # Pull in the non-production credentials for the build to use
+          - group: stacks-credentials-nonprod-kv #ensono-sp-creds
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: done

File: src/deploy-aca.yml [hunk 28]
```diff
@@ -0,0 +1,314 @@
...
+                          env:
+                            TF_FILE_LOCATION: /app/src/terraform
+                            TF_BACKEND_INIT: "key=$(tf_state_key),container_name=$(tf_state_container),storage_account_name=$(tf_state_storage),resource_group_name=$(tf_state_rg)"
+                            TF_BACKEND_DESTROY_PLAN: "-input=false,-out=\"destroy\""
+                            TF_VAR_name_environment: ${{ stage.environment_shortname }}
+                            CLOUD_PLATFORM: "$(cloud_platform)"
+                            ARM_TENANT_ID: $(azure-tenant-id)
+                            ARM_SUBSCRIPTION_ID: $(azure-subscription-id)
+                            ARM_CLIENT_ID: $(azure-client-id)
+                            ARM_CLIENT_SECRET: $(azure-client-secret)
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: done

File: src/deploy-aca.yml [hunk 29]
```diff
@@ -0,0 +1,314 @@
...
+                          env:
+                            TF_FILE_LOCATION: /app/src/terraform
+                            TF_BACKEND_INIT: "key=$(tf_state_key),container_name=$(tf_state_container),storage_account_name=$(tf_state_storage),resource_group_name=$(tf_state_rg)"
+                            TF_BACKEND_PLAN: "-input=false,-out=\"tfplan\""
+                            TF_VAR_name_environment: ${{ stage.environment_shortname }}
+                            CLOUD_PLATFORM: "$(cloud_platform)"
+                            ARM_TENANT_ID: $(azure-tenant-id)
+                            ARM_SUBSCRIPTION_ID: $(azure-subscription-id)
+                            ARM_CLIENT_ID: $(azure-client-id)
+                            ARM_CLIENT_SECRET: $(azure-client-secret)
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: done

File: src/deploy-aca.yml [hunk 30]
```diff
@@ -0,0 +1,314 @@
...
+  - stage: Release
+    dependsOn:
+      - ${{ if ne(variables['Build.SourceBranch'], 'refs/heads/master') }}:
+          - NonProd
+      - ${{ else }}:
+          - Prod
+    condition: and(succeeded(), or(eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(${{ parameters.pre_release }}, true)))
+    variables:
+      # TODO: Credentials should come from another group. `amido-stacks-github-credentials` are the old creds
+      - group: amido-stacks-github-credentials   #amido-stacks-webapp
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: done

File: src/deploy-aca.yml [hunk 31]
```diff
@@ -0,0 +1,314 @@
...
+                        TF_VAR_name_component: $(domain)
+                        TF_VAR_name_environment: ${{ stage.environment_shortname }}
+                        TF_VAR_stage: ${{ stage.environment_shortname }}
+                        TF_VAR_location: $(region)
+                        TF_VAR_acr_name:  $(container_registry_name)
+                        TF_VAR_acr_rg:  $(registry_resoruce_group)
+                        TF_VAR_acae_name: $(container_environment_name)
+                        TF_VAR_image_name: "$(container_registry_server)/$(appname)"
+                        TF_VAR_image_tag: $(version_number)
+                        TF_VAR_container_app_name: $(container_app_name)
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml:182]: This is app specific variables, we are not targeting Stacks-CLI to output app specific variables .

File: deploy/tests/controls/resource_group.rb
[the remark this answers is not in the document]
Comment (satenderrathee) [reply deploy/tests/controls/resource_group.rb:1]: Done

File: src/deploy-aca.yml [hunk 32]
```diff
@@ -0,0 +1,314 @@
...
+  - template: ../cicd/build/azDevOps/azure/pipeline-vars.yml  
+  - template: aca-vars.yml
+  - name: appname
+    value: "frontendapp"
+  - name: tf_state_key
+    value: $(domain)_$(appname)
+  # - template: agent-config-vars.yml
+
+  # - name: CHEF_LICENSE
+  #   value: accept-silent
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: Done

File: src/deploy-aca.yml [hunk 33]
```diff
@@ -0,0 +1,314 @@
...
+                    #     TF_VAR_name_project: $(project)
+                    #     TF_VAR_name_component: $(domain)
+                    #     TF_VAR_name_environment: ${{ stage.environment_shortname }}
+                    #     TF_VAR_stage: ${{ stage.environment_shortname }}
+                    #     TF_VAR_location: $(region)
+                    #     TF_VAR_acr_name:  $(container_registry_name)
+                    #     TF_VAR_acae_name: $(container_environment_name)
+                    #     TF_VAR_image_name: "$(container_registry_server)/$(appname)"
+                    #     TF_VAR_image_tag: $(version_number)
+                    #     TF_VAR_container_app_name: $(container_app_name)
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply src/deploy-aca.yml]: Testing included now

File: cicd/build/taskctl/tasks.yaml [hunk 34]
```diff
@@ -0,0 +1,178 @@
...
+
+  tests:infra:run:
+    context: infratests
+    description: Execute infrastructure tests against deployed Cloud resources
+    command:
+      - |
+        Invoke-Terraform -Workspace -Arguments $env:TF_VAR_name_environment -Path $env:TF_FILE_LOCATION
+        Invoke-Terraform -Output -Path $env:TF_FILE_LOCATION | /app/cicd/build/scripts/Set-EnvironmentVars.ps1 -prefix "TFOUT" -key "value" -passthru | ConvertTo-Yaml | Out-File -Path ${PWD}/inspec_inputs.yml
+        Add-Content -Path ${PWD}/inspec_inputs.yml -Value "location: ${env:TF_VAR_location}"
+        Invoke-Inspec -execute -arguments "--input-file ${PWD}/inspec_inputs.yml" -path $env:INSPEC_FILES
```
[the remark this answers is not in the document]
Comment (satenderrathee) [reply cicd/build/taskctl/tasks.yaml:126]: I appreciate the insight from your Redis feature tasks. For now, I'd like to keep the current structure as it aligns well with our existing patterns and meets our needs effectively. I'll keep your approach in mind for future optimizations.

**Comment by satenderrathee** [comment]
> Good overall work, some rethinking required around directory structure and redundant variable files. Lots of left over comments to remove and a few lock files.
> 
> Needs some decent documentation within the repo to support the feature outlining: Why are we doing this feature? How do you use this feature? What is the pipeline doing? What are the tests doing?

Thanks Dan for the review. I have actioned the review comments now.
I don't think the repo documentation is in the scope of current story but I will add the task for it and will action this in a separate PR.

File: deploy/tests/controls/acae.rb
Comment (danEnsono) [inline deploy/tests/controls/acae.rb:1]: Typo in the name?

**Review by danEnsono — CHANGES_REQUESTED** [review]
All good except one of the test names, fix that and good to go!

File: deploy/tests/controls/acae.rb
Comment (satenderrathee) [reply deploy/tests/controls/acae.rb:1]: updated the name :)

**Review by danEnsono — APPROVED** [review]
Good to go!

**Review by danEnsono — APPROVED** [review]

**Pull request merged**